### PR TITLE
Convert home mofo link to protocol button-icon

### PR DESCRIPTION
**Pull request by janbrasna**
## One-line summary

Changes the complicated styling+imagery used to denote an external link to the newly available `*-button-icon-end` in v19.1.0 and using just base protocol SVG iconography.

## Significant changes and points to review

There may be another convention on how to colorise the (black) SVG icons when not inlined but used as `` or css backgrounds, so please _do tell_ if you mind the filters hack.

(Also note the inverted glyph color is not 100% white, but rather off-grey, which I coincidentally find more pleasing to the eye, leaving the max contrast to the actual link text 🤷… Going for almost-white is possible chaining more filters with rotations, contrasts, sepias etc. that make the value non-intuitive / non-manageable later though.)

## Issue / Bugzilla link

_not an issue at all_

## Testing

http://localhost:8000/en-US/  "Learn about the Mozilla Foundation ⎋"

Screenshot 2024-04-19 at 16 40 27

### CODE REVIEW

File: media/css/mozorg/home/home-new.scss [hunk 1]
```diff
@@ -188,30 +188,17 @@ header {
     }
 
     .c-foundation {
-        a {
+        a.mzp-c-button[rel~="external"] {
             display: inline-flex;
             align-items: center;
```
Comment (reemhamz) [inline media/css/mozorg/home/home-new.scss:194]: **Nit + suggestion:** The icon is shrinking to smaller sizes when the window is smaller. Adding a `flex-shrink: 0` to the `span` element holding the `svg` should fix this!

<kbd>Screenshot 2024-04-23 at 4 23 56 PM</kbd>

**Review by reemhamz — APPROVED** [review]
Just a tiny thing to patch up and it should be good to go. Good job implementing this!
r+wc

File: media/css/mozorg/home/home-new.scss (hunk 1 above)
Comment (janbrasna) [reply media/css/mozorg/home/home-new.scss:194]: You're absolutely right, fixed.🤦 How did I miss this? Thanks!! Really appreciate you triple quadruple checking even the most trivial changes after me turning knobs and touching buttons to see what explodes where;D

I originally wanted to get rid of the inline-flex altogether, but the way the base `button-icon-end` leaves the icon "dangling" on an empty new line, inlined/treated like text for longer CTAs like this, doesn't look very appealing. So I kept the idea of the original author, to improve the relation between text and icon, and it's way better.

I'm actually trying this button as a proof-of-concept for https://github.com/mozilla/bedrock/pull/14407#pullrequestreview-2006649547 that could benefit from it too, with so many longer CTAs suffering from the orphaned icons on new lines.

**Comment by stephaniehobson** [comment]
Needs a rebase 🥺

**Comment by janbrasna** [comment]
@stephaniehobson thanks, done, that was my own conflict! I had a hunch these two line edits won't be too far apart;)

**Pull request merged**